code/modules/mob/living/emote.dm

```diff
@@ -125,13 +125,33 @@
 		if(H.Togglewings())
 			addtimer(CALLBACK(H,TYPE_PROC_REF(/mob/living/carbon/human, Togglewings)), wing_time)
 
+/datum/emote/living/flap/select_message_type(mob/user, intentional)
+	. = ..()
+	var/wings = FALSE
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.dna && H.dna.species)
+			if((H.dna.features["wings"] != "None") || H.dna.features["moth_wings"] != "None")
```

Comment:
These DNA features are inherited from your preferences, and can be set even when you are not a moth (and more often than not are just a random moth wing selection). As such checking for this is only useful if you have already checked if they even have the wing organ.

Reply:
A good example of what you can do is given in /datum/emote/living/carbon/human/wing (line 122 in emote.dm)

Reply:
guh
dumb
will change that